src/generated/cloud/essentialcontacts/v1/src/model.rs

```diff
@@ -638,160 +638,289 @@ impl wkt::message::Message for SendTestMessageRequest {
+        match self {
+            Self::Unspecified => std::option::Option::Some("VALIDATION_STATE_UNSPECIFIED"),
+            Self::Valid => std::option::Option::Some("VALID"),
+            Self::Invalid => std::option::Option::Some("INVALID"),
+            Self::UnknownValue(u) => u.0.name(),
         }
     }
+}
 
-    /// Creates an enum value from the value name.
-    pub fn from_str_name(name: &str) -> std::option::Option<Self> {
-        match name {
-            "VALIDATION_STATE_UNSPECIFIED" => {
-                std::option::Option::Some(Self::VALIDATION_STATE_UNSPECIFIED)
-            }
-            "VALID" => std::option::Option::Some(Self::VALID),
-            "INVALID" => std::option::Option::Some(Self::INVALID),
-            _ => std::option::Option::None,
-        }
+impl std::default::Default for ValidationState {
+    fn default() -> Self {
+        use std::convert::From;
+        Self::from(0)
+    }
+}
+
+impl std::fmt::Display for ValidationState {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::result::Result<(), std::fmt::Error> {
+        wkt::internal::display_enum(f, self.name(), self.value())
     }
 }
 
 impl std::convert::From<i32> for ValidationState {
     fn from(value: i32) -> Self {
-        Self::new(value)
+        match value {
+            0 => Self::Unspecified,
+            1 => Self::Valid,
+            2 => Self::Invalid,
+            _ => Self::UnknownValue(validation_state::UnknownValue(
+                wkt::internal::UnknownEnumValue::Integer(value),
+            )),
+        }
     }
 }
 
-impl std::default::Default for ValidationState {
-    fn default() -> Self {
-        Self::new(0)
+impl std::convert::From<&str> for ValidationState {
+    fn from(value: &str) -> Self {
+        use std::string::ToString;
+        match value {
+            "VALIDATION_STATE_UNSPECIFIED" => Self::Unspecified,
+            "VALID" => Self::Valid,
+            "INVALID" => Self::Invalid,
+            _ => Self::UnknownValue(validation_state::UnknownValue(
+                wkt::internal::UnknownEnumValue::String(value.to_string()),
+            )),
+        }
+    }
+}
+
+impl serde::ser::Serialize for ValidationState {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match self {
+            Self::Unspecified => serializer.serialize_i32(0),
+            Self::Valid => serializer.serialize_i32(1),
+            Self::Invalid => serializer.serialize_i32(2),
+            Self::UnknownValue(u) => u.0.serialize(serializer),
+        }
+    }
+}
+
+impl<'de> serde::de::Deserialize<'de> for ValidationState {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        deserializer.deserialize_any(wkt::internal::EnumVisitor::<ValidationState>::new(
+            ".google.cloud.essentialcontacts.v1.ValidationState",
```

Comment:
hmm so we only use this name when printing error messages... arguably not worth it, but fine.